### PR TITLE
gossamer host api tests: add test_blake2_128

### PR DESCRIPTION
**Pull request by noot**
Hey, I've added a function for calling `rtm_blake2_128` which works:

```
$ ./gossamer-adapter host-api  --function test_blake2_128 --input "Horizontal"
DBUG[06-18|17:37:47] [NewRuntime]                             runtimeCtx="{storage:0xc0001a8510 allocator:0xc00020c000 keystore:0xc0000a7d60}"
TRCE[06-18|17:37:47] [ext_malloc] executing...                size=10
TRCE[06-18|17:37:47] [ext_blake2_128] executing... 
TRCE[06-18|17:37:47] [ext_malloc] executing...                size=16
TRCE[06-18|17:37:47] [ext_free] executing...                  addr=32
TRCE[06-18|17:37:47] [ext_malloc] executing...                size=20
TRCE[06-18|17:37:47] [ext_free] executing...                  addr=56
[17/1114032]0x7f42040a0050
```

let me know if the logging causes an issue, I can update gossamer to not print logs when using packages as a library

### CODE REVIEW

File: test/adapters/gossamer/go.sum [hunk 1]
```diff
@@ -22,6 +22,14 @@ github.com/ChainSafe/go-schnorrkel v0.0.0-20200405005733-88cbf1b4c40d h1:nalkkPQ
 github.com/ChainSafe/go-schnorrkel v0.0.0-20200405005733-88cbf1b4c40d/go.mod h1:URdX5+vg25ts3aCh8H5IFZybJYKWhJHYMTnf+ULtoC4=
 github.com/ChainSafe/gossamer v0.1.1-0.20200615204208-759e66ed3a06 h1:2/I89UcNymTXgF23KHA2Zp5q4oGjEH/Tb7LMbRnY/YU=
 github.com/ChainSafe/gossamer v0.1.1-0.20200615204208-759e66ed3a06/go.mod h1:xRh5l13AXR0ZFRMewA2lyjaDvuqsWRAckxuEs2xaN9I=
+github.com/ChainSafe/gossamer v0.1.1-0.20200615204448-9497e408e971 h1:nU4N70WDZ1lDYNbZ/F7jC813yLbMW4msFMz9qxvcjVQ=
+github.com/ChainSafe/gossamer v0.1.1-0.20200615204448-9497e408e971/go.mod h1:xRh5l13AXR0ZFRMewA2lyjaDvuqsWRAckxuEs2xaN9I=
+github.com/ChainSafe/gossamer v0.1.1-0.20200617213134-ec61e5518d03 h1:O7w2DRq9hPBL/xYSHAv6xUCKdcseSgQSmqtQ1VILTeo=
+github.com/ChainSafe/gossamer v0.1.1-0.20200617213134-ec61e5518d03/go.mod h1:xRh5l13AXR0ZFRMewA2lyjaDvuqsWRAckxuEs2xaN9I=
+github.com/ChainSafe/gossamer v0.1.1-0.20200618150322-dd7eb885c9f9 h1:5nppdf0/g/wxd2VEaMd4YHucI7LmjAfUhKs9jTjQNCc=
+github.com/ChainSafe/gossamer v0.1.1-0.20200618150322-dd7eb885c9f9/go.mod h1:xRh5l13AXR0ZFRMewA2lyjaDvuqsWRAckxuEs2xaN9I=
```
Comment (FlorianFranzen) [inline test/adapters/gossamer/go.sum]: You could probably get of the first these by running `go mod tidy`. Not that it matters much.

File: test/adapters/gossamer/host_api.go [hunk 2]
```diff
@@ -20,21 +20,21 @@
...
 	"github.com/ChainSafe/gossamer/lib/trie"
-	"github.com/ChainSafe/gossamer/dot/state"
 
 	database "github.com/ChainSafe/chaindb"
 )
 
-
-const RELATIVE_WASM_ADAPTER_PATH = "adapters/wasm-legacy/target/release/wbuild/wasm-adapter-legacy/wasm_adapter_legacy.compact.wasm"
-
+var RELATIVE_WASM_ADAPTER_PATH = "../../adapters/wasm-legacy/target/release/wbuild/wasm-adapter-legacy/wasm_adapter_legacy.compact.wasm"
```
Comment (FlorianFranzen) [inline test/adapters/gossamer/host_api.go]: `runtest.jl` ensures that all adapters are run from the test folder. Therefore, for the adapter to work in the testsuite, this change has to be reverted. Alternatively, one could also statically link in the byte array `WASM_BINARY` from the `libwasm_adapter_legacy.rlib` as it contains the runtime as well. (Maybe through cbindgen?)

**Review by FlorianFranzen — APPROVED** [review]
Thank you for getting this started.

File: test/adapters/gossamer/go.sum (hunk 1 above)
Comment (noot) [reply test/adapters/gossamer/go.sum]: good point, will do!

File: test/adapters/gossamer/host_api.go (hunk 2 above)
Comment (noot) [reply test/adapters/gossamer/host_api.go]: I'll change it back then. I was testing it from the `gossamer` directory

**Comment by FlorianFranzen** [comment]
@noot Great, thanks. The CI looks much happier now: 
```
Test Summary:   | Pass  Fail  Total
Host API Legacy |   76   164    240
```
The 76 passes also contains some false positives, as "unknown" functions do not cause the adapter to return an error at the moment, but that is ok for now.
I just merged it (had to do it on the command line). Thanks again.

**Pull request closed without merging**